psqlgraph/mocks.py

```diff
@@ -0,0 +1,471 @@
+        pass
+
+
+class EnumRand(Randomizer):
+    def __init__(self, values):
+        super(EnumRand, self).__init__()
+        self.values = values
+
+    def random_value(self, override=None):
+        if override in self.values:
+            return override
+        return random.choice(self.values)
+
+
+class NumberRand(Randomizer):
+    def __init__(self, type_def):
+        super(NumberRand, self).__init__()
+        self.minimum = type_def.get('minimum', 0)
+        self.maximum = type_def.get('maximum', 10000)
+
+    def random_value(self, override=None):
+        if isinstance(override, int) and \
+                self.minimum <= override <= self.maximum:
+            return override
+        return random.randrange(self.minimum, self.maximum + 1)
+
+
+class StringRand(Randomizer):
+    def __init__(self, type_def):
+        super(StringRand, self).__init__()
+        if type_def.get('format') == 'date-time':
+            self.pattern = '201[0-9]-0[1-9]-[0-2][1-9]T00:00:00'
+        else:
+            self.pattern = type_def.get('pattern', '[A-Za-z0-9]{32}')
+
+    def random_value(self, override=None):
+        if isinstance(override, str):
+            return override
+        return rstr.xeger(self.pattern)
+
+
+class BooleanRand(Randomizer):
+    def random_value(self, override=None):
+        if isinstance(override, bool):
+            return override
+        return random.choice([True, False])
+
+
+class TypeRandFactory(object):
+    @staticmethod
+    def get_randomizer(type_def):
+        _type = type_def.get('type')
+        if isinstance(_type, list):
+            _type = _type[0]
+
+        if _type in ['object', 'array']:
+            raise ValueError('Resolve relationships outside of this factory')
+
+        if _type in ['integer', 'number', 'float']:
+            return NumberRand(type_def)
+
+        if _type in ['boolean']:
+            return BooleanRand()
+
+        return StringRand(type_def)
+
+
+class PropertyFactory(object):
+    def __init__(self, properties):
+        self.properties = properties
+        self.type_factories = {}
+        for name in properties:
+            try:
+                self.type_factories[name] = self.resolve_type(
+                    self.properties.get(name, {})
+                )
+            except ValueError as ve:
+                # print(
+                #     "Property: '{}' is most likely a relationship. Error: {}"
+                #     "".format(name, ve)
+                # )
+                pass
+
+    def create(self, name, override=None):
+        """
+        Given a property name, create a property-name, property-value pair.
+        If override is provided, use override value as property-value
+
+        :param name: property name
+        :type name: String
+        :param override: property value
+        :type override: type
+        :return: Tuple(str, type)
+        """
+        return name, self.type_factories[name].random_value(override)
```

Comment:
This assumes that `self.type_factories[name]` is defined. Shouldn't you make sure it actually is?